lightstep/sdk/metric/aggregator/histogram/histogram.go

```diff
@@ -0,0 +1,352 @@
+// 1ms to 100s).  See the derivation here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exponential-histogram-aggregation
+const DefaultMaxSize = 160
+
+// MinSize is the smallest reasonable configuration, which is small
+// enough to contains the entire normal flowing point range at
+// MinScale.
+const MinSize = 2
+
+type (
+	// buckets stores counts for measurement values in the range
+	// (0, +Inf).
+	buckets struct {
+		// backing is a slice of nil, []uint8, []uint16, []uint32, or []uint64
+		backing interface{}
+
+		// indexBase is index of the 0th position in the
+		// backing array, i.e., backing[0] is the count associated with
+		// indexBase which is in [indexStart, indexEnd]
+		indexBase int32
+
+		// indexStart is the smallest index value represented
+		// in the backing array.
+		indexStart int32
+
+		// indexEnd is the largest index value represented in
+		// the backing array.
+		indexEnd int32
+	}
+
+	// highLow is used to establish the maximum range of bucket
+	// indices needed, in order to establish the best value of the
+	// scale parameter.
+	highLow struct {
+		low  int32
+		high int32
+	}
+)
+
+var (
+	_ aggregator.Methods[int64, Int64]     = Methods[int64, number.Int64Traits, Int64]{}
+	_ aggregator.Methods[float64, Float64] = Methods[float64, number.Float64Traits, Float64]{}
+
+	_ aggregation.Histogram = &Int64{}
+	_ aggregation.Histogram = &Float64{}
+)
+
+func NewFloat64(cfg aggregator.HistogramConfig, values ...float64) *Float64 {
+	return newHist[float64, number.Float64Traits](cfg, values...)
+}
+
+func NewInt64(cfg aggregator.HistogramConfig, values ...int64) *Int64 {
+	return newHist[int64, number.Int64Traits](cfg, values...)
+}
+
+type Option func(aggregator.HistogramConfig) aggregator.HistogramConfig
+
+func WithMaxSize(maxSize int32) Option {
+	return func(cfg aggregator.HistogramConfig) aggregator.HistogramConfig {
+		cfg.MaxSize = maxSize
+		return cfg
+	}
+}
+
+func NewConfig(opts ...Option) aggregator.HistogramConfig {
+	var cfg aggregator.HistogramConfig
+	for _, opt := range opts {
+		cfg = opt(cfg)
+	}
+	return cfg
+}
+
+func newHist[N number.Any, Traits number.Traits[N]](cfg aggregator.HistogramConfig, values ...N) *State[N, Traits] {
+	var methods Methods[N, Traits, State[N, Traits]]
+
+	state := &State[N, Traits]{}
+
+	methods.Init(state, aggregator.Config{
+		Histogram: cfg,
+	})
+
+	for _, val := range values {
+		methods.Update(state, val)
+	}
+	return state
+}
+
+// Sum implements aggregation.Histogram.
+func (h *State[N, Traits]) Sum() number.Number {
+	var t Traits
+	return t.ToNumber(h.sum)
+}
+
+// Count implements aggregation.Histogram.
+func (h *State[N, Traits]) Count() uint64 {
+	return h.count
+}
+
+// Scale implements aggregation.Histogram.
+func (s *State[N, Traits]) Scale() int32 {
+	if s.data.count == s.data.zeroCount {
+		// all zeros! scale doesn't matter, use zero.
+		return 0
```

Comment:
I added a branch and comment to `incrementIndexBy()`.
```
	if incr == 0 {
		// Skipping a bunch of work for 0 increment.  This
		// happens when merging sparse data, for example.
		// This also happens UpdateByIncr is used with a 0
		// increment, means it can be safely skipped.
		return highLow{}, true
	}
```

Reply:
Thanks!